packages/documentation/pages/usage/components/button.vue

```diff
@@ -81,6 +81,18 @@
     <KtButton type="primary" icon="edit" helpText="This is an icon button"/>
     <KtButton type="primary" icon="edit" label="Right Icon Button" iconPosition="right"/>
     ```
+
+    ## `isToggle`
+
+    *  For buttons that are **toggleable**, and can have two different status: "ON" or "OF"
+    *  This prop is only valid for buttons of type **"default"** or type **"text"**.
+    *  Apart from **isToggle** prop it needs **toggleStatus** prop that can be either **on** or **of**.
+
+     <div class="element-example white">
+    	<KtButton type="default" :isToggle="true" :toggleStatus="toggleDefaultStatus" @click="(event) => onToggleDefaultClick(event)" class="mr-4">Default Toggle button</KtButton>
```

Comment:
@Juli0GT I'd much rather you have a special event for toggling not just the click

basically

if the prop for toggling on and off is called x , you have an event called update:x

with vue3 this will allow us to do v-model:x so that whoever wants to use a toggle button in the user app can just v-model:x easily and doesn't need an explicit handler

the click can remain as is but not be used

Reply:
my 2nd opinion from an API perspective (usage perspective)

in order to achieve the functionality you want, you ended up having to add 2 props (isToggle and toggleStatus) 

instead

the API just exposes `toggleStatus` and emits `update:toggleStatus`

IF the user defines toggleStatus then it's a toggle button

if the user doesn't define toggleStatus then it's not

so instead of having two props for the feature you have ONE "implication" prop (implication because the status already implies that this is a toggle button)